packages/@uppy/xhr-upload/src/index.js

```diff
@@ -502,29 +524,48 @@ module.exports = class XHRUpload extends Plugin {
   }
 
   uploadFiles (files) {
-    const actions = files.map((file, i) => {
+    const promises = files.map((file, i) => {
       const current = parseInt(i, 10) + 1
       const total = files.length
 
       if (file.error) {
         return () => Promise.reject(new Error(file.error))
```

Comment:
#todo - we have this error in master too, this should be `return Promise.reject(new Error(file.error))` instead, otherwise we're getting `.then` is not a function on request fail&retry, with aws-s3 plugin:

![image](https://user-images.githubusercontent.com/7578559/63789337-96e8a700-c910-11e9-89dc-6e82a1eb2c7c.png)

Reply:
`() => ` on master is correct, but this PR changes the other return values to Promises instead of functions, so I should change this too. thanks!